contracts/dtoken/src/interest_rate_model.rs

```diff
@@ -82,6 +102,26 @@ impl InterestRateModel{
         self.account_supply_interest.insert(&account, &interest);
     }
 
+    pub fn calculate_interest_on_borrow(&mut self, account: AccountId, borrow_rate: Ratio, total_borrow: Balance) {
+        let current_block_height = block_height();
+        if self.get_borrow_block_by_user(account.clone()) == 0 {
+            self.set_borrow_block_by_user(account, current_block_height);
+        } else {
+            let accrued_rate = total_borrow * borrow_rate * (current_block_height - self.get_borrow_block_by_user(account.clone())) as u128 / RATIO_DECIMALS;
+            self.set_borrow_block_by_user(account.clone(), current_block_height);
+            self.set_borrow_interest_by_user(account, accrued_rate);
+        }
+    }
+
+    pub fn calculate_interest_on_repay(&mut self, account: AccountId, borrow_rate: Ratio, total_borrow: Balance) -> Balance {

```

Comment:
@dariacherniak the same as for supply interest we can use only one function for recalculating interest on borrow